startup.sh

```diff
@@ -79,20 +138,37 @@ __     __        _  __         _   _       _        ____  ___
 EOF
 tput sgr0
 
+# Options for rebuilding the data directory
+if [[ $SKIP_DATA_CHECK == "false" ]]; then
+  check_data_age
+fi
+
+if [[ $REBUILD_DATA == "true" ]]; then
+  rebuild_data
+fi
+
 # Running generate scripts in docker avoids having to install their
 # dependencies on the host.
-docker build -t verify-local-startup .
+docker build -t verify-local-startup \
+--build-arg USER_ID=$(id -u) \
+--build-arg GROUP_ID=$(id -g) .
 
 # Inlining the following block of commands to docker run rather than putting
 # them in their own script to avoid an extra level of bash indirection
 docker run -t -v "$script_dir:/verify-local-startup/" verify-local-startup '
 set -e
 if ! test -d data; then
   generate/hub-dev-pki.sh
+  chown -R verify:staff data
+  chown verify:staff *.env
```

Comment:
Why is this needed? What if you run this on a non-GDS machine?
What if the `verify` user doesn't exist?

Reply:
Isn't it concerning that in order to run a bunch of Java apps the scripts needs to use `chown`? I'm wondering if this scripts is getting too "clever" or is trying to do too many things. It doesn't seem to me that this should be necessary

Reply:
So its actually a work around the issue of Docker... Whenever Docker does anything on Linux with the host operating system it creates files as root because it runs as root.  This means we can't do basics like easily remove the data directory when it gets stale after say a 2 week period.